kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -85,6 +88,17 @@ protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, final
         }
         final RequestFrame frame;
         if (decodeRequest) {
+            short highestProxyVersion = apiKey.latestVersion(true);
```

Comment:
I wondered about `enableUnstableLastVersion = true`.  I wondered if this could mean a production environment could start using unstable versions spontaneously.  I decided that `true` is okay.  The client and broker would need to also have the same configuration for this is to be true. 

I think what you have done is okay.

Reply:
here's the commit where it was [added](https://github.com/kroxylicious/kroxylicious/pull/389/commits/7f2e4e51eac41a0fe645cc2e59974398432b0066)

I feel like I discussed it somewhere, but can't dig it up. The commit says:

> 3. `ApiMessageType#highestSupportedVersion()` has been removed/replaced
with `ApiMessageType$highestSupportedVersion(boolean enableUnstableLastVersion)`
I've made Kroxylicious support the unstable last version as it should be the
decision of the backing broker whether to accept unstable versions, the
proxy should handle them.

So yeah, the thought was by default the proxy shouldn't interfere if the client/broker want to use unstable versions.